flocker/apiclient/test/test_client.py

```diff
@@ -144,6 +144,39 @@ def got_result(dataset):
             d.addCallback(got_result)
             return d
 
+        def test_delete(self):
+            """
+            ``delete_dataset`` removes a dataset.
+            """
+            dataset_id = uuid4()
+
+            d = self.assert_creates(self.client, primary=self.node_1,
+                                    maximum_size=DATASET_SIZE,
+                                    dataset_id=dataset_id)
+            d.addCallback(
+                lambda _: self.client.delete_dataset(dataset_id))
+
+            def got_result(dataset):
+                expected_removed = Dataset(
+                    dataset_id=dataset_id, primary=self.node_1,
+                    maximum_size=DATASET_SIZE
+                )
+                self.assertEqual(expected_removed, dataset)
```

Comment:
:question: Unless you're using a single high-level assertion method that internally does multiple assertions, it's better to have only one assertion in each unit test.